internal/watcher/watcher.go

```diff
@@ -80,13 +81,13 @@ type serviceImpl struct {
 
 var _ Service = (*serviceImpl)(nil)
 
-func New(ctx context.Context, c *Config) (*serviceImpl, error) {
+func New(ctx context.Context, c *Config, logger logr.Logger) (*serviceImpl, error) {
```

Comment:
I think that there is a pattern in this project where the logger is listed first, or second when there's a context, then the rest of parameters. That said, I think it should be: `New(ctx context.Context, logger logr.Logger, c *Config)`.